crates/uv-client/src/middleware.rs

```diff
@@ -45,3 +46,18 @@ impl Middleware for OfflineMiddleware {
         ))
     }
 }
+
+#[derive(thiserror::Error, Debug)]
+pub(crate) enum CertificateError {
+    #[error(transparent)]
+    Io(#[from] std::io::Error),
+    #[error(transparent)]
+    Reqwest(#[from] reqwest::Error),
+}
+
+/// Return the `Identity` from the provided file.
+pub(crate) fn read_identity(ssl_client_cert: &OsStr) -> Result<Identity, CertificateError> {
```

Comment:
This feel like kind of a weird place for this to live. Any particular reason you put it in the middleware module?

Reply:
I didn't think about it too much. Just thought its specific for reqwest usage so I saw it as middleware. Should I move these to a new `tls.rs` file?

Reply:
Moved in d9f04b83843b5154099d4a53b5435df1fe873962

Reply:
Ah I would have just put it in the `base_client` module since it's just used for configuration. This works though, thanks!